src/main/java/com/smartling/connector/eloqua/sdk/client/DynamicContentClient.java

```diff
@@ -1,95 +1,104 @@
+     {
+         return executeCall(dynamicContentApi -> dynamicContentApi.getDynamicContent(EloquaApi.Depth.PARTIAL, id));
+     }
+
+     public DynamicContent createDynamicContent(DynamicContent dynamicContentToCreate)
+     {
+         return executeCall(dynamicContentApi -> dynamicContentApi.createDynamicContent(dynamicContentToCreate));
+     }
+
+     public void updateDynamicContent(Long id, DynamicContent dynamicContent)
+     {
+         executeCall(dynamicContentApi -> dynamicContentApi.updateDynamicContent(id, dynamicContent));
+     }
+
+     public Void deleteDynamicContent(final long id)
+     {
+         return executeCall(dynamicContentApi -> dynamicContentApi.deleteDynamicContent(id));
+     }
+
+     public Elements<DynamicContent> searchForDynamicContent(final String name)
+     {
+         return executeCall(dynamicContentApi -> dynamicContentApi.searchForDynamicContent(EloquaApi.Depth.PARTIAL, "[name =] \"" + name + '"'));
+     }
+
+    public void saveTranslatedRule(final long id, final long fieldId, final String locale, final String translatedContent)
+    {
+        DynamicContent dynamicContent = getDynamicContent(id);
+
+        final Rule rule = createNewRuleForDynamicContent(fieldId, locale, translatedContent, dynamicContent);
+
+        dynamicContent.getRules().add(rule);
+
+        updateDynamicContent(dynamicContent.getId(), dynamicContent);
+    }
+
+    public static Rule createNewRuleForDynamicContent(final long fieldId, final String locale, final String translatedContent, final DynamicContent dynamicContent)
+    {
+        final ContentSection contentSection = new ContentSection();
+        contentSection.setType(dynamicContent.getDefaultContentSection().getType());
+        contentSection.setDepth(dynamicContent.getDefaultContentSection().getDepth());
+        contentSection.setScope(dynamicContent.getDefaultContentSection().getScope());
+        contentSection.setSize(dynamicContent.getDefaultContentSection().getSize());
+        contentSection.setFolderId(dynamicContent.getDefaultContentSection().getFolderId());
+        contentSection.setContentHtml(translatedContent);
+        contentSection.setName(locale);
+
+        final Condition condition = new Condition();
+        condition.setValue(locale);
+        condition.setOperator("equal");
+        condition.setType("TextValueCondition");
+
+        final Criterium criterium = new Criterium();
+        criterium.setType("ContactFieldCriterion");
+        criterium.setFieldId(fieldId);
+        criterium.setCondition(condition);
+
+        final Rule rule = new Rule();
+        rule.setType("DynamicContentRule");
+        rule.setDepth("complete");
+        rule.setCriteria(Collections.singletonList(criterium));
+        rule.setContentSection(contentSection);
+        rule.setStatement("1111");//could be anything but required by API :D
+        return rule;
+    }
+
+    public DynamicContent copyDynamicContent(long id, String name)
```

Comment:
I don't think we need this method and tests. We don't clone dynamic content asset because we create new rules in the same asset. So i would suggest to remove this and keed code base as small as possible.

Reply:
We support two kinds of DynamicContent, and that which is not single-file requires copying